### PR TITLE
Add aggregation fields to views, sessions, users

### PR DESCRIPTION
**Pull request by rlh1994**
## Description

The next step for the most common request and use case for custom models is aggregations of (base) events this run records for that view/session. This adds the same risk we did with passthrough fields, which it is easy to make mistakes with this from users and we have no control over the SQL they write to ensure it's correct and error free. This approach at least ensures that we only allow certain aggregation types and the grouping id which means we can be sure there aren't any 1:many joins. Users, as always, has to re-aggregate from sessions the same as passthrough fields. 

I am still of the opinion that this isn't _that_ much easier than just adding a custom model and joining to e.g. views this run, especially as to do anything like count events of a specific type you have to write a `case when` statement, but if it helps a few people it isn't much work to keep in the package.

Example var:
`snowplow__view_aggregations: [{"type": "sum", "field": "case when event_name = 'button_click' then 1 else 0 end", "alias": "num_ button_clicks" ]`

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
https://snplow.atlassian.net/browse/PE-5992

## Checklist
- [x] 🎉 I have verified that these changes work locally
- [ ] 💣 Is your change a breaking change?
- [ ] 📖 I have updated the CHANGELOG.md

### Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [x] 📓 internal package docs (ymls, macros, readme, if applicable)
- [ ] 📕 I have raised a [Snowplow documentation](https://github.com/snowplow/documentation) PR if applicable (Link here if required)
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Review by georgewoodhead — APPROVED** [review]
LGTM! ⛏️ bump changlog version

File: CHANGELOG [hunk 1]
```diff
@@ -1,3 +1,14 @@
+snowplow-unified 0.2.0 (2024-02-XX)
```
[the remark this answers is not in the document]
Comment (rlh1994) [reply CHANGELOG:1]: I'll do that in the next PR because I do not want the tests to run again 🥲

File: CHANGELOG [hunk 2]
```diff
@@ -1,3 +1,14 @@
...
+---------------------------------------
+## Summary
+XXX
+
+## Features
+- Add new passthrough aggregations to the views, sessions, and users table, enabled using `snowplow__view/session/user_aggregations`
```
Comment (agnessnowplow) [inline CHANGELOG:7]: Are you confident in that the '/' in the var name won't break something down the line?

**Review by agnessnowplow — APPROVED** [review]
lgtm2

**Pull request merged**